src/bt_peer_connection.cpp

```diff
@@ -1660,7 +1662,9 @@ namespace libtorrent
 #endif
 				return;
 			}
-			bool ul = detail::read_uint8(recv_buffer.begin) != 0;
+			char const* pos1 = recv_buffer.begin();
```

Comment:
yeah, these are kind of the messy uses of const_interval. I added a separate set of read_*() and write_*() functions that advances the start position of the span<>. I forget which namespace their in off hand